editor/src/components/navigator/navigator-item/layout-icon.tsx

```diff
@@ -23,11 +34,40 @@ export function layoutIconTestIdForEntry(navigatorEntry: NavigatorEntry): string
   return `layout-icn-${navigatorEntryToKey(navigatorEntry)}`
 }
 
+export function isZeroSizedDiv(elementPath: ElementPath, metadata: ElementInstanceMetadataMap) {
+  const bounds = MetadataUtils.getFrameInCanvasCoords(elementPath, metadata)
+  if (bounds == null || isInfinityRectangle(bounds)) {
+    return false
+  }
+
+  const isElementDiv =
+    optionalMap(
+      (i) => MetadataUtils.isDiv(i),
+      MetadataUtils.findElementByElementPath(metadata, elementPath),
+    ) ?? false
+
+  return isZeroSizedElement(bounds) && isElementDiv
+}
+
+const isZeroSizedDivSelector = createCachedSelector(
+  metadataSelector,
+  (_: MetadataSubstate, x: ElementPath) => x,
+  (metadata, elementPath) => {
+    return isZeroSizedDiv(elementPath, metadata)
+  },
+)((_, x) => EP.toString(x))
+
 export const LayoutIcon: React.FunctionComponent<React.PropsWithChildren<LayoutIconProps>> =
   React.memo((props) => {
     const { elementWarnings, color, warningText: propsWarningText, navigatorEntry } = props
     const { iconProps, isPositionAbsolute } = useLayoutOrElementIcon(navigatorEntry)
 
+    const isZeroSized = useEditorState(
```

Comment:
Minor, name picked to not clash with existing things like `isZeroSizedDiv`.
```suggestion
    const zeroSizedDiv = useEditorState(
```